src/frontend/src/lib/stores/transactions.store.ts

```diff
@@ -0,0 +1,61 @@
+import type { BtcTransactionUi } from '$btc/types/btc';
+import type { IcTransactionUi } from '$icp/types/ic';
+import { initCertifiedStore, type CertifiedStore } from '$lib/stores/certified.store';
+import type { CertifiedData } from '$lib/types/store';
+import type { TokenId } from '$lib/types/token';
+import { nonNullish } from '@dfinity/utils';
+
+export type CertifiedTransaction<T> = CertifiedData<T>;
+
+export type TransactionsData<T> = CertifiedTransaction<T>[];
+
+export interface TransactionsStore<T> extends CertifiedStore<TransactionsData<T>> {
+	prepend: (params: { tokenId: TokenId; transactions: CertifiedTransaction<T>[] }) => void;
+	append: (params: { tokenId: TokenId; transactions: CertifiedTransaction<T>[] }) => void;
+	cleanUp: (params: { tokenId: TokenId; transactionIds: string[] }) => void;
+}
+
+export const initTransactionsStore = <
```

Comment:
I assume this is "basically" a refactoring of existing code and we will use it for IC in a separate PR right?

Reply:
Exactly.